### PR TITLE
feat(ui): default sizes to the available disk space

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Use the available disk space as the default value for the sizes in the storage forms.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View the storage tab for a machine.
- In the available storage table click 'Add partition'
- The form should open and the size field should match the disk's free space.
- Click 'Add partition'.
- Tick the new partition and click 'Create volume group' and submit.
- Open the action menu for the volume group and click 'Add logical volume'.
- The form should open and the size field should match the volume group's free space.

## Fixes

Fixes: #2375.

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddLogicalVolume/AddLogicalVolume.tsx [hunk 1]
```diff
@@ -107,7 +108,7 @@ export const AddLogicalVolume = ({
           mountOptions: "",
           mountPoint: "",
           name: initialName,
-          size: "",
+          size: formatBytes(disk.available_size, "GB").value,
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddLogicalVolume/AddLogicalVolume.tsx:111]: The second argument is actually supposed to be the unit of the first argument, which in this case is `"B"`, so this should be `formatBytes(disk.available_size, "B", { convertTo: "GB" }).value`.

File: ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddPartition/AddPartition.tsx [hunk 2]
```diff
@@ -94,14 +94,15 @@ export const AddPartition = ({
...
+        allowUnchanged
         buttons={FormCardButtons}
         cleanup={machineActions.cleanup}
         errors={errors}
         initialValues={{
           fstype: "",
           mountOptions: "",
           mountPoint: "",
-          partitionSize: "",
+          partitionSize: formatBytes(disk.available_size, "GB").value,
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddPartition/AddPartition.tsx:105]: Same here, `formatBytes(disk.available_size, "B", { convertTo: "GB" }).value`

**Review by Caleb-Ellis — CHANGES_REQUESTED** [review]
`formatBytes` isn't quite being used correctly here, though admittedly it's a bit confusing.

By default it converts to a unit where the value would be between 1-1000, so if you pass in 10000KB it'll return 10MB, if you pass in 0.1GB it'll return 100MB, etc. You have to include the `convertTo` option if you want to convert the value to a specific unit.

File: ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddLogicalVolume/AddLogicalVolume.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddLogicalVolume/AddLogicalVolume.tsx]: Done

File: ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddPartition/AddPartition.tsx (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AddPartition/AddPartition.tsx]: Done

**Review by Caleb-Ellis — APPROVED** [review]

**Pull request merged**